packages/cli-kit/src/private/node/session/exchange.ts

```diff
@@ -167,18 +167,32 @@ interface TokenRequestResult {
 }
 
 function tokenRequestErrorHandler(error: string) {
-  if (error === 'invalid_grant') {
+  let errorMessage = error
+  if (errorMessage === 'invalid_grant') {
     // There's an scenario when Identity returns "invalid_grant" when trying to refresh the token
     // using a valid refresh token. When that happens, we take the user through the authentication flow.
     return new InvalidGrantError()
   }
-  if (error === 'invalid_request') {
+  if (errorMessage === 'invalid_request') {
     // There's an scenario when Identity returns "invalid_request" when exchanging an identity token.
     // This means the token is invalid. We clear the session and throw an error to let the caller know.
     return new InvalidRequestError()
   }
+  if (errorMessage === 'invalid_target') {
```

Comment:
```ts
function tokenRequestErrorHandler(error: string) {
  if (error === 'invalid_grant') {
    // There's an scenario when Identity returns "invalid_grant" when trying to refresh the token
    // using a valid refresh token. When that happens, we take the user through the authentication flow.
    return new InvalidGrantError()
  }
  if (error === 'invalid_request') {
    // There's an scenario when Identity returns "invalid_request" when exchanging an identity token.
    // This means the token is invalid. We clear the session and throw an error to let the caller know.
    return new InvalidRequestError()
  }
  if (error === 'invalid_target') {
  // add a comment here
    return new InvalidTargetError(INVALID_TARGET_ERROR_MESSAGE)
  }
  // eslint-disable-next-line @shopify/cli/no-error-factory-functions
  return new AbortError(error)
}
```

Reply:
^ can we try keeping these patterns as is?

Reply:
Ah ok thanks for the context! That's a great observation that I wasn't aware of :) 

As long as the message is in a constant, I'm happy :)

Reply:
I agree with extracting the message to a constant. And maybe there's no need to create a new error, you could just do something like `return new AbortError(INVALID_TARGET_ERROR_MESSAGE)`.